src/classes/BDI_FieldMappingCustomMetadata.cls

```diff
@@ -49,11 +70,65 @@ public with sharing class BDI_FieldMappingCustomMetadata implements BDI_FieldMap
 
-        return new Map<String,String>();
+        String fieldMappingSetName = dis.Default_Data_Import_Field_Mapping_Set__c;
+        Data_Import_Field_Mapping_Set__mdt diFieldMappingSet;
+        Data_Import_Field_Mapping__mdt[]  diFieldMappings;
+        Data_Import_Object_Mapping__mdt diObjectMapping;
+
+        if(fieldMappingSetName != null){
+            diFieldMappingSet = [SELECT id, DeveloperName, 
+                                    Data_Import_Object_Mapping_Set__c,
+                                    Data_Import_Object_Mapping_Set__r.DeveloperName    
+                                FROM Data_Import_Field_Mapping_Set__mdt
+                                WHERE DeveloperName =: fieldMappingSetName LIMIT 1];
+
+            if(diFieldMappingSet != null && diFieldMappingSet.Data_Import_Object_Mapping_Set__c != null){
+                
+                diFieldMappings = [SELECT DeveloperName,
+                                        Data_Import_Field_Mapping_Set__c,
+                                        Required__c,
+                                        Source_Field_API_Name__c,
+                                        Target_Field_API_Name__c,
+                                        Target_Object_Mapping__c,
+                                        Target_Object_Mapping__r.Object_API_Name__c
+                            FROM Data_Import_Field_Mapping__mdt
+                            WHERE Target_Object_Mapping__r.Data_Import_Object_Mapping_Set__r.DeveloperName =: diFieldMappingSet.Data_Import_Object_Mapping_Set__r.DeveloperName
+                                AND Target_Object_Mapping__r.DeveloperName =: dataImportObjectName];
+
+                if(diFieldMappings != null){
+                    for(Data_Import_Field_Mapping__mdt difm : diFieldMappings){
+
+                        String sourceFN = difm.Source_Field_API_Name__c;
+                        String targetFN = difm.Target_Field_API_Name__c;
+
+                        //Overwriting whatever was passed in for the object api name from the cmt record
+                        targetObjectName = difm.Target_Object_Mapping__r.Object_API_Name__c;
+
+                        //If we are not in the npsp namespace then remove the namespace from all relevant field references
+                        if(ns != 'npsp'){
+                            sourceFN = UTIL_Namespace.StrTokenRemoveNSPrefixNpspOnly(sourceFN);
+                            targetFN = UTIL_Namespace.StrTokenRemoveNSPrefixNpspOnly(targetFN);
+                        }
+
+                        dataImportFieldToTargetField.put(sourceFN,targetFN);
+                    }
+                }
+            } 
+        }
+
+        return dataImportFieldToTargetField;
     }
+
+    /*******************************************************************************************************
+    * @description our exception object for Field Level Security errors.
+    */
+    public class FLSException extends Exception {}
```

Comment:
With the move of the FLS  checks to the service I think we can pull this exception type from this class as well (unless we think we may want to bring it back in for the future)